addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -590,22 +590,21 @@ function factory(dependencies) {
         /**
          * @private
          * @param {Object} param0
-         * @param {integer[]} param0.message_ids
          * @param {boolean} param0.starred
+         * @param {integer} param0.starred_counter
+         * @param {Object} data
          */
-        _handleNotificationPartnerToggleStar({ message_ids = [], starred }) {
+        _handleNotificationPartnerToggleStar({ starred, starred_counter, data }) {
             const starredMailbox = this.messaging.starred;
-            for (const messageId of message_ids) {
-                const message = this.messaging.models['mail.message'].findFromIdentifyingData({
-                    id: messageId,
-                });
+            for (const messageData of data) {
+                const message = this.messaging.models['mail.message'].insert(
+                    this.messaging.models['mail.message'].convertData(messageData)
+                );
                 if (!message) {
                     continue;
                 }
```

Comment:
Since `message` is no longer retrieved using `findFromIdentifyingData`, this shouldn't be needed, as message should always be defined.